src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java

```diff
@@ -260,7 +287,8 @@ public MagicNumberCheck() {
 
     @Override
     public void visitToken(DetailAST ast) {
-        if ((!ignoreAnnotation || !isChildOf(ast, TokenTypes.ANNOTATION))
+        if (!skipAnnotationArgs(ast)
+                && !skipAnnotationDefaults(ast)
```

Comment:
double negative in expressions ... can we avoid this ?

lets make it as:
```
 if (shouldTestAnnotationArgs(ast)
                && shouldTestAnnotationDefaults(ast)
```

Reply:
done